lib/Mojo/ByteStream.pm

```diff
@@ -48,8 +48,8 @@ sub secure_compare { Mojo::Util::secure_compare ${shift()}, shift }
 sub size { length ${$_[0]} }
 
 sub split {
-  my ($self, $pattern) = @_;
-  return Mojo::Collection->new(map { $self->new($_) } split $pattern, $$self);
+  my ($self, $pattern, $limit) = @_;
+  return Mojo::Collection->new(map { $self->new($_) } split $pattern, $$self, $limit//0);
```

Comment:
Using `//0` also seems pointless. What do we gain by explicitly specifying a default?

Reply:
If you changed this a little bit you would get away without an extra line or default value.
```perl
  my ($self, $pat) = (shift, shift);
  return Mojo::Collection->new(map { $self->new($_) } split $pat, $$self, @_);
```

Reply:
How do you run perltidy?  The -b option is incompatible with -pbp.  Do you just do something like this:
```
perltidy file > file.tdy && mv file.tdy file
```

Reply:
How do you run perltidy?  The -b option is incompatible with -pbp.  Do you just do something like this:
```
perltidy file > file.tdy && mv file.tdy file
```

Reply:
I didn't use @_ because it doesn't work, and I supplied a default value of 0 because split causes a warning when an uninitialized value is passed.

The best I could do while not increasing line count is:
```
  my ($self, $pattern, $limit) = @_;
  return c(map { $self->new($_) } split $pattern, $$self, $limit // 0);
```

That requires importing 'c' from Mojo::Collection, of course.

After that I'm fresh out of ideas.  :man_shrugging:  Feel free to close this PR if that is unlikely to pass the vote.

Reply:
I didn't expect all `split` arguments to be special cases. Guess the best you can do is this:
```perl
sub split {
  my ($self, $pat, $lim) = (shift, shift, shift // 0);
  return Mojo::Collection->new(map { $self->new($_) } split $pat, $$self, $lim);
}
```